src/marqo/vespa/vespa_client.py

```diff
@@ -110,6 +114,9 @@ def download_application(self) -> str:
         Returns:
             Path to the downloaded application
         """
+        if wait_for_application_convergence:
+            self.wait_for_application_convergence()
```

Comment:
As noted I suggest a check and error if hasn't converged, rather than wait

Reply:
wait for 1 second